packet_packer.go

```diff
@@ -20,6 +21,8 @@ type packedPacket struct {
 }
 
 type packetPacker struct {
+	mutex sync.Mutex
```

Comment:
I'm not a big fan of this mutex here. We're more and more breaking up our concept of not having concurrent code apart from the session, and this makes stuff much harder to reason about. In particular, to understand this mutex, I need to know which entry paths into this class exist that may be concurrent.

So, I would propose two options:

- Either make packing the ConnectionClose sufficiently independant from normal packing that we don't need it anymore (could we e.g. pass the controlFrames as param to packPacket?).
- Fully protect everything in this class with the mutex (e.g. RetransmitNonForwardSecurePacket), so that there's no hidden knowledge required to extend this in the future.